src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Linux.cs

```diff
@@ -128,10 +128,17 @@ internal static ProcessInfo CreateProcessInfo(Interop.procfs.ParsedStat procFsSt
         {
             int pid = procFsStat.pid;
 
+            // Get long process name if possible, otherwise use a fall back method.
+            string procName = Path.GetFileName(Process.GetExePath(pid));
+            if (String.IsNullOrEmpty(procName))
```

Comment:
under what conditions can this happen?

Reply:
I don't have the backgroud of investigation I think it's better left @2E0PGS to answer.

Reply:
If you see the issue for this PR there is some background for this solution: https://github.com/dotnet/corefx/issues/34437